app/liquid_tags/youtube_tag.rb

```diff
@@ -29,19 +34,23 @@ def parse_id(input)
     input_no_space
   end
 
+  def valid_id?(id)
+    id.match?(/\A[a-zA-Z0-9_-]{11}((\?t=)?(\d{1,}h?)?(\d{1,2}m)?(\d{1,2}s)?){5,11}?\Z/)
+  end
+
   def translate_start_time(id)
     time = id.split("?t=")[-1]
-    time_hash = {
-      h: time.scan(/\d+h/)[0]&.delete("h").to_i,
-      m: time.scan(/\d+m/)[0]&.delete("m").to_i,
-      s: time.scan(/\d+s/)[0]&.delete("s").to_i
-    }
-    time_in_seconds = (time_hash[:h] * 3600) + (time_hash[:m] * 60) + time_hash[:s]
-    "#{id.split('?t=')[0]}?start=#{time_in_seconds}"
-  end
+    return "#{id.split('?t=')[0]}?start=#{time}" if time.match?(/\A\d+\Z/)
 
-  def valid_id?(id)
-    id =~ /\A[a-zA-Z0-9_-]{11}((\?t=)?(\d{1}h)?(\d{1,2}m)?(\d{1,2}s)?){5,11}?\Z/
+    time_elements = time.split(/[a-z]/)
+    time_markers = time.split(/\d+/)[1..]
+
+    seconds = 0
+    time_markers.each_with_index do |m, i|
+      seconds += MARKER_TO_SECONDS_MAP.fetch(m, 0) * time_elements[i].to_i
+    end
```

Comment:
I love this change. Using the hash to map units to seconds is such a great move!

Reply:
I must give honor to whom it is due: @jeremyf helped with this approach 🙏🏾